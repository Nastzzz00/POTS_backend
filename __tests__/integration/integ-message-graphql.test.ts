import gql from 'graphql-tag';
import { createTestClient } from 'apollo-server-testing';
import { constructTestServer } from '../__utils';

const { server }: any = constructTestServer({
  context: {},
});

describe('Tests', () => {
  //Queries
  it('login as admin ', async () => {
    const MessageQuery = gql`
      query i($credential: CredentialInput!) {
        login(credential: $credential) {
          message
          loggedIn
          userLevel
        }
      }
    `;
    const { query } = createTestClient(server);
    const res = await query({
      query: MessageQuery,
      variables: {
        credential: {
          username: 'bmNhcnBpbw==',
          password: 'MTIz',
        },
      },
    });

    expect(res).toMatchSnapshot();
  });

  it('login as supplier ', async () => {
    const MessageQuery = gql`
      query i($credential: CredentialInput!) {
        login(credential: $credential) {
          message
          loggedIn
          userLevel
        }
      }
    `;
    const { query } = createTestClient(server);
    const res = await query({
      query: MessageQuery,
      variables: {
        credential: {
          username: 'YXNhbm95',
          password: 'MTIz',
        },
      },
    });

    expect(res).toMatchSnapshot();
  });

  it('login as supplier ', async () => {
    const MessageQuery = gql`
      query i($credential: CredentialInput!) {
        login(credential: $credential) {
          message
          loggedIn
          userLevel
        }
      }
    `;
    const { query } = createTestClient(server);
    const res = await query({
      query: MessageQuery,
      variables: {
        credential: {
          username: 'Something Something',
          password: 'Dota 2',
        },
      },
    });

    expect(res).toMatchSnapshot();
  });
});
