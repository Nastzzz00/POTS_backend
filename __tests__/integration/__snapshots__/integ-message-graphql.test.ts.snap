// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests login as admin  1`] = `
Object {
  "data": Object {
    "login": Object {
      "loggedIn": true,
      "message": "Admin Logged In",
      "userLevel": "Admin",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests login as supplier  1`] = `
Object {
  "data": Object {
    "login": Object {
      "loggedIn": true,
      "message": "Supplier Logged In",
      "userLevel": "Supplier",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests login as supplier  2`] = `
Object {
  "data": Object {
    "login": Object {
      "loggedIn": false,
      "message": "Wrong Credentials",
      "userLevel": "none",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
